src/main/java/dev/misei/html2pdf/config/SecurityConfig.java

```diff
@@ -0,0 +1,18 @@
+package dev.misei.html2pdf.config;
+
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+
+@Configuration
+public class SecurityConfig {
+
+    @Bean
+    public FilterRegistrationBean<ApiKeyFilter> apiKeyFilter() {
+        FilterRegistrationBean<ApiKeyFilter> registrationBean = new FilterRegistrationBean<>();
```

Comment:
I like it. Check this piece of code with a similar approach. Take a look how the filter is inserted.:

> 

```@Bean
    public SecurityFilterChain securityFilterChain(HttpSecurity http,
                                                   UserDetailsService userDetailsService,
                                                   JwtTokenProvider jwtTokenProvider,
                                                   AuthenticationManager authenticationManager) throws Exception {
        http
                .csrf(AbstractHttpConfigurer::disable)
                .cors(AbstractHttpConfigurer::disable)
                .userDetailsService(userDetailsService)
                .sessionManagement((policy) -> policy.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
                .formLogin(AbstractHttpConfigurer::disable)
                .logout(configurer -> configurer.logoutUrl("/api/public/auth/logout"))
                .headers(HeadersConfigurer::cacheControl)
                .authorizeHttpRequests(registry -> registry
                        .anyRequest().permitAll())
                .addFilterBefore(new JwtTokenFilter(jwtTokenProvider, authenticationManager), UsernamePasswordAuthenticationFilter.class);

        return http.build();
    }